server/src/main/java/org/opensearch/gateway/remote/routingtable/RemoteIndexRoutingTable.java

```diff
@@ -9,92 +9,106 @@
+    ) {
+        super(clusterUUID, compressor, null);
+        this.index = indexRoutingTable.getIndex();
         this.indexRoutingTable = indexRoutingTable;
+        this.term = term;
+        this.version = version;
     }
 
     /**
      * Reads data from inputStream and creates RemoteIndexRoutingTable object with the {@link IndexRoutingTable}
-     * @param inputStream input stream with index routing data
-     * @param index index for the current routing data
-     * @throws IOException exception thrown on failing to read from stream.
+     * @param blobName name of the blob, which contains the index routing data
+     * @param clusterUUID UUID of the cluster
+     * @param compressor Compressor object
      */
-    public RemoteIndexRoutingTable(InputStream inputStream, Index index) throws IOException {
-        try {
-            try (BufferedChecksumStreamInput in = new BufferedChecksumStreamInput(new InputStreamStreamInput(inputStream), "assertion")) {
-                // Read the Table Header first and confirm the index
-                IndexRoutingTableHeader indexRoutingTableHeader = new IndexRoutingTableHeader(in);
-                assert indexRoutingTableHeader.getIndexName().equals(index.getName());
-
-                int numberOfShardRouting = in.readVInt();
-                IndexRoutingTable.Builder indicesRoutingTable = IndexRoutingTable.builder(index);
-                for (int idx = 0; idx < numberOfShardRouting; idx++) {
-                    IndexShardRoutingTable indexShardRoutingTable = IndexShardRoutingTable.Builder.readFrom(in);
-                    indicesRoutingTable.addIndexShard(indexShardRoutingTable);
-                }
-                verifyCheckSum(in);
-                indexRoutingTable = indicesRoutingTable.build();
-            }
-        } catch (EOFException e) {
-            throw new IOException("Indices Routing table is corrupted", e);
-        }
+    public RemoteIndexRoutingTable(String blobName, String clusterUUID, Compressor compressor) {
+        super(clusterUUID, compressor, null);
+        this.index = null;
+        this.term = -1;
+        this.version = -1;
+        this.blobName = blobName;
     }
 
     public IndexRoutingTable getIndexRoutingTable() {
         return indexRoutingTable;
     }
 
-    /**
-     * Writes {@link IndexRoutingTable} to the given stream
-     * @param streamOutput output stream to write
-     * @throws IOException exception thrown on failing to write to stream.
-     */
+    public Index getIndex() {
+        return index;
+    }
+
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        return new BlobPathParameters(List.of(indexRoutingTable.getIndex().getUUID()), "");
+    }
+
+    @Override
+    public String getType() {
+        return INDEX_ROUTING_TABLE;
+    }
+
+    @Override
+    public String generateBlobFileName() {
+        return String.join(
+            DELIMITER,
+            INDEX_ROUTING_TABLE_PREFIX,
```

Comment:
File names should be of format `%s_%s `and not `%s-%s`. Lets keep the format consistent.
For different usecases, lets keep variables separate to avoid confusion.